tripal/t4d8
# Update Registration Form

# New Feature

Issue tripal/t4d8#24 

## Description

Implemented `\Drupal\tripal\Form\Register::validateForm` and `\Drupal\tripal\Form\Register::submitForm`. This should complete the registration form functionality. All functionality I have added here is based off of Tripal 3.x's [tripal.registration.inc](https://github.com/tripal/tripal/blob/7.x-3.x/tripal/includes/tripal.registration.inc). The form should now operate the same way it did in Tripal 3.x.

The data that the T4 registration form sends to [tripal.info](https://tripal.info/) in [`\Drupal\tripal\Form\Register::tripal_registration_remote_submit`](https://github.com/par12005/t4d8/blob/d8-tv4-issue24-registrationform/tripal/src/Form/Register.php#L343) is the same data and same format as the T3 registration form did in [`tripal_registration_remote_submit`](https://github.com/tripal/tripal/blob/7.x-3.x/tripal/includes/tripal.registration.inc#L316). Also cleaned up the code a bit.

## Testing

 - Clear your cache
 - Navigate to `admin/tripal/register`
 - Fill out the form
 - Verify that the appropriate updates have been made in the `public.key_value` table in the database

## Additional Notes

I'm not sure how `tripal.info` handles data as it comes in via registration forms, but it may be a good idea to either specify a new endpoint, or indicate the Tripal version somewhere in the registration info.
